controller/src/main/java/io/pravega/controller/store/index/ZKHostIndex.java

```diff
@@ -93,12 +94,13 @@ public ZKHostIndex(CuratorFramework client, String hostRoot, Executor executor)
     @Override
     public CompletableFuture<List<String>> getEntities(final String hostId) {
         Preconditions.checkNotNull(hostId);
-        return getChildren(getHostPath(hostId));
+        String hostPath = getHostPath(hostId);
+        return sync(hostPath).thenCompose(v -> getChildren(hostPath));
     }
 
     @Override
     public CompletableFuture<Set<String>> getHosts() {
-        return getChildren(hostRoot).thenApply(list -> list.stream().collect(Collectors.toSet()));
+        return sync(hostRoot).thenCompose(v -> getChildren(hostRoot)).thenApply(list -> list.stream().collect(Collectors.toSet()));
```

Comment:
Is it really necessary to sync here?

Reply:
This gets the list of hosts. During bootstrap sweep performed for failover we first call getHosts and then we iterate over hosts to call getEntities. So its called during bootstrap and its good to do a sync before calling it.